internal/service/alertconfiguration/data_source_alert_configuration_test.go

```diff
@@ -2,32 +2,27 @@ package alertconfiguration_test
 
 import (
 	"fmt"
-	"os"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
-	"go.mongodb.org/atlas-sdk/v20231115007/admin"
 )
 
 func TestAccConfigDSAlertConfiguration_basic(t *testing.T) {
 	var (
-		alert          = &admin.GroupAlertsConfig{}
-		dataSourceName = "data.mongodbatlas_alert_configuration.test"
-		orgID          = os.Getenv("MONGODB_ATLAS_ORG_ID")
-		projectName    = acc.RandomProjectName()
+		projectID = acc.ProjectIDExecution(t)
 	)
 
-	resource.ParallelTest(t, resource.TestCase{
+	resource.Test(t, resource.TestCase{
```

Comment:
is this intentional?

Reply:
I guess it is based on the lines below :) maybe I missed it in the description but I am not understanding why we are not making it parallel anymore

Reply:
My understanding is that before, each test was creating a project, so now that the different tests use the same one, the tests will fail if run in parallel

Reply:
yes, I changed alert_configuration and database_user test to be serial on purpose.  The reason is that they are resources common to the project.  
For instance one test for alert_configuration plural datasource creates 2 alerts and ask if there are two, it will fail if other tests are creating others.  
alert_notification creates like 20 projects and each one tests only like 5 seconds, so now they will create only 1 project and will take 2 minutes to run all.
however tests than don't change common project elements (which are the most expensive to run like the ones using clusters) can continue working on parallel.

Reply:
as an example in this PR, these resources are reusing projects and running in parallel:
cloud_backup_schedule
cloud_backup_snapshot
global_cluster_config
search_index
stream_connection

Reply:
> alert_notification creates like 20 projects and each one takes only like 5 seconds, so now they will create only 1 project and will take 2 minutes to run all.
Is it possible to bring more concrete data here before we proceed? And same for `database_user`? My rationale is that "a bit of slow down here, a bit of slow down there, ..." becomes a big slow down and development experience might be affected.

I took a small extraction of https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/8273381828/job/22637080203 and I can see that tests finished after 20-ish seconds:
```
--- PASS: TestAccConfigRSAlertConfiguration_withoutRoles (16.76s)
--- PASS: TestAccConfigRSAlertConfiguration_importIncorrectId (17.03s)
--- PASS: TestAccConfigRSAlertConfiguration_withVictorOps (17.15s)
--- PASS: TestAccConfigRSAlertConfiguration_withPagerDuty (17.66s)
--- PASS: TestAccConfigRSAlertConfiguration_withOpsGenie (17.66s)
--- PASS: TestAccConfigRSAlertConfiguration_withoutOptionalAttributes (18.64s)
--- PASS: TestAccConfigRSAlertConfiguration_importConfigNotifications (18.88s)
--- PASS: TestAccConfigRSAlertConfiguration_importBasic (20.31s)
--- PASS: TestAccConfigRSAlertConfiguration_importPagerDuty (21.25s)
--- PASS: TestAccConfigRSAlertConfiguration_withMatchers (22.23s)
--- PASS: TestAccConfigRSAlertConfiguration_basic (24.72s)
--- PASS: TestAccConfigRSAlertConfiguration_updatePagerDutyWithNotifierId (24.72s)
--- PASS: TestAccConfigRSAlertConfiguration_withNotifications (24.82s)
--- PASS: TestAccConfigDSAlertConfigurations_totalCount (25.03s)
--- PASS: TestAccConfigRSAlertConfiguration_withMetricUpdated (25.79s)
--- PASS: TestAccConfigRSAlertConfiguration_withThresholdUpdated (26.49s)
--- PASS: TestAccConfigDSAlertConfiguration_withOutput (11.65s)
--- PASS: TestAccConfigDSAlertConfiguration_withThreshold (10.68s)
--- PASS: TestAccConfigDSAlertConfigurations_basic (14.24s)
--- PASS: TestAccConfigDSAlertConfigurations_withOutputTypes (15.80s)
```

I can't quantify how much it really took since they were ran in parallel, but if we imagine to make that sequential wouldn't it be more than 2 mins?

Reply:
both alert_configuration and database_user are in "config" test group,  in your link it takes 3m5s

in this [PR](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/8280595555/job/22657662093?pr=2007) it takes 2m42s.

So even in serial it takes less time now  :-)

Most of time in your log is creating the project, that's why they take in the 20s.

Anyway I'm going to try what @AgustinBettati said of only runnning serially the plural ds test, that should wotk.

Reply:
@marcosuma @andreaangiolillo after @AgustinBettati idea to run all in parallel except plural ds there are the new PR times 🎉 

ok  	github.com/mongodb/terraform-provider-mongodbatlas/internal/service/databaseuser	23.636s
ok  	github.com/mongodb/terraform-provider-mongodbatlas/internal/service/alertconfiguration	33.800s

Reply:
ok it actually did improve and your rationale seems right to me: avoiding creating the project all the times actually contributes to saving time.

Reply:
Two follow-ups coming from this PR and conversation (not urgent, but worth a discussion):
1. couldn't we also convert plural tests to be parallel by changing the expectations? (e.g. remove the "check number of alerts equal n" in favour of "check alert with 'this' and 'this' name exists)
2. In n months (just random time in the future) when someone will see the code base and see some tests as running in parallel and some sequential, what do you think it could go wrong? Is there a risk that without proper clarification on the code base they might think it's an error and start investigating why?

Reply:
1 - yes, that is a good idea, let me think about it
2 - we already do have a mix of parallel and serial (before this PR).  let me see 1 if we can do the ds plural serial so we minimize exceptions.